llmfoundry/utils/warnings.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2024 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+
+class VersionedDeprecationWarning(DeprecationWarning):
+    """A custom deprecation warning class that includes version information.
+
+    Attributes:
+        message (str): The deprecation message describing why the feature is deprecated.
+        after_version (str): The version after which the feature will be deprecated.
```

Comment:
The wording here isn't totally clear to me. Should it be "Feature X is deprecated and will be removed after version Y"?